QuizApp/Scenes/QuizPage/ViewController/ProgressView.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  ProgressView.swift
+//  QuizApp
+//
+//  Created by Tamuna Kakhidze on 18.11.24.
+//
+
+import UIKit
+
+// MARK: - ProgressView
+final class ProgressView: UIView {
+    
+    // MARK: - UI Components and Properties
+    private let blueProgressView: UIView = {
+        let progressView = UIView()
+        progressView.translatesAutoresizingMaskIntoConstraints = false
+        progressView.backgroundColor = CustomColors.blueSecondaryDefault
+        progressView.layer.cornerRadius = Constants.Sizing.radius
+        return progressView
+    }()
+    
+    private var progressWidthConstraint: NSLayoutConstraint?
+    var quizQuestionCount: Int?
+    
+    // MARK: - Lifecycle
+    init() {
+        super.init(frame: .zero)
+        initialize()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - Setup
+    private func initialize() {
+        setAppearance()
+        setupViewHierarchy()
+        setConstraints()
+    }
+    
+    private func setAppearance() {
+        translatesAutoresizingMaskIntoConstraints = false
+        backgroundColor = CustomColors.neutralLighterGrey
+        layer.cornerRadius = Constants.Sizing.radius
+    }
+    
+    private func setupViewHierarchy() {
+        addSubview(blueProgressView)
+    }
+    
+    private func setConstraints() {
+        progressWidthConstraint = blueProgressView.widthAnchor.constraint(
```

Comment:
Why don't you move this into the constraints activation clause, just curious

Reply:
it is activated later in function